src/Files.App/Actions/Content/Run/RunWithPowershellAction.cs

```diff
@@ -0,0 +1,47 @@
+﻿using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.DependencyInjection;
+using Files.App.Commands;
+using Files.App.Contexts;
+using Files.App.Extensions;
+using Files.App.Helpers;
+using Files.App.Shell;
+using Files.Backend.Helpers;
+using System.Threading.Tasks;
+
+namespace Files.App.Actions
+{
+	internal class RunWithPowershellAction : ObservableObject, IAction
+	{
+		public IContentPageContext context = Ioc.Default.GetRequiredService<IContentPageContext>();
+
+		public bool IsExecutable => context.SelectedItem is not null &&
+			FileExtensionHelpers.IsPowerShellFile(context.SelectedItem.FileExtension);
+
+		public string Label => "RunWithPowerShell".GetLocalizedResource();
+
+		public string Description => "TODO: Need to be described.";
+
+		public RichGlyph Glyph => new("\uE756");
+
+		public RunWithPowershellAction()
+		{
+			context.PropertyChanged += Context_PropertyChanged;
+		}
+
+		public async Task ExecuteAsync()
+		{
+			Win32API.RunPowershellCommand(context.ShellPage?.SlimContentPage?.SelectedItem.ItemPath, false);
```

Comment:
Do you want to spin up a helper for this?

Reply:
What do you mean?